packages/fetchai/connections/p2p_libp2p/utils/utils.go

```diff
@@ -378,10 +464,53 @@ func FetchAIAddressFromPublicKey(publicKey string) (string, error) {
 	if err != nil {
 		return addr, err
 	}
-	addr, err = bech32.Encode("fetch", fiveBitsChar)
+	addr, err = bech32.Encode(prefix, fiveBitsChar)
 	return addr, err
 }
 
+// EthereumAddressFromPublicKey get wallet address from hex encoded secp256k1 public key
+// format from: https://github.com/fetchai/agents-aea/blob/master/aea/crypto/ethereum.py#L330
+// check also: https://github.com/ethereum/go-ethereum/blob/master/crypto/crypto.go#L263
+func EthereumAddressFromPublicKey(publicKey string) (string, error) {
+	var addr string
+	var err error
+	hexBytes, err := hex.DecodeString(publicKey[2:])
```

Comment:
see above, worth adding in comment that `0x` is dropped

Reply:
same, I believe it should be transparent, as dropping `0x` is related to hex decoder in use